### ISSUE TITLE
Error handler relies on locale setting

### ISSUE THREAD

**Issue by Pointer666** [php-amqplib/php-amqplib#557]
The error handler does only work with an english locale setting.
Example:
```
setlocale(LC_MESSAGES,'de_DE');
```
could result in the following error message.
```
fwrite(): send of 3728 bytes failed with errno=11 Die Ressource ist zur Zeit nicht verfügbar
```
this is not handled by the error handler and an exception will be thrown

https://github.com/php-amqplib/php-amqplib/blob/dfd3694a86f1a7394d3693485259d4074a6ec79b/PhpAmqpLib/Wire/IO/StreamIO.php#L307-L336

**Comment by lukebakken** [comment]
We would review a PR to address this. Basically we would have to compare error code or some non-string value for those two cases.

**Comment by lukebakken** [comment]
Addressed by #630
